lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -511,6 +563,49 @@ class TraceProcessor {
     return {lcp, invalidated};
   }
 
+  /**
+   * @param {{frame: string, url: string, parent?: string}[]} frames
+   * @return {Map<string, string>}
+   */
+  static resolveRootFrames(frames) {
```

Comment:
objects in this array have a property `frame`, which is confusing (`const frame = frames[0]; frame.frame ...`). I see L573 accounts for this by calling each item `frameData`.

Perhaps another way to handle it is to map the `FrameCommittedInBrowser` events to `{id: string, url: string}` (set `id` to `data.frame`), and then you can use `frame` for the singular entry in this array.